functions/fish_prompt.fish

```diff
@@ -846,6 +847,85 @@ function __bobthefish_prompt_rubies -S -d 'Display current Ruby information'
     echo -ns $ruby_glyph $ruby_version ' '
 end
 
+function __bobthefish_prompt_golang -S -d 'Display current Go information'
+    [ "$theme_display_go" = 'no' ]
+    and return
+    
+    set -l go_version
+    if type -fq asdf
+        set -l asdf_golang_version (asdf current golang 2>/dev/null)
+        or set -l asdf_golang_version "na"
+        
+        set -l asdf_go_sdk_version (asdf current go-sdk 2>/dev/null)
+        or set -l asdf_go_sdk_version "na"
+
+        set -l _asdf_plugin
+        set -l asdf_go_version
+        set -l asdf_provenance
+        if [ "$asdf_golang_version" != "na" ]
+            echo "$asdf_golang_version" | read  _asdf_plugin asdf_go_version asdf_provenance
+        else if [ "$asdf_go_sdk_version" != "na" ]
+            echo "$asdf_go_sdk_version" | read  _asdf_plugin asdf_go_version asdf_provenance
+        end
+
+        set go_version $asdf_go_version
+    end
+
+
+    set -l cwd (pwd)
+    set -l dir (pwd)
+    set -l found_gomod 0
+    set -l no_go_installed 0
+    set -l gomod_version "0"
+    set -l gomod_file
+
+    # find the closest go.mod
+    while not test "$dir" = "/"
+        set gomod_file "$dir/go.mod"
+
+        if test -f "$gomod_file"
+            set found_gomod 1
+            cat "$gomod_file" | grep "^go\ " | read _gomod gomod_version 
+            break
+        end
+
+        cd $dir/..
```

Comment:
we don't need to actually `cd` around the filesystem. We just want to find and read a file, right?

check out how the `hg` implementation looks for a `.hg` directory. it's almost exactly what we want to do here.

https://github.com/oh-my-fish/theme-bobthefish/blob/332f23abd7a095d5b2c024a061af7b890a4f0c20/functions/fish_prompt.fish#L230-L242